tests/app/dao/notification_dao/test_letter_api_notification_dao.py

```diff
@@ -5,62 +5,84 @@
     NOTIFICATION_SENDING,
     KEY_TYPE_TEST,
     KEY_TYPE_NORMAL,
-    LETTER_TYPE
+    LETTER_TYPE,
+    LETTERS_AS_PDF,
+    EMAIL_TYPE
 )
 from app.dao.notifications_dao import dao_set_created_live_letter_api_notifications_to_pending
 
-from tests.app.db import create_notification, create_service, create_template
+from tests.app.db import create_notification, create_service, create_template, create_job
 
 
 def test_should_only_get_letter_notifications(
-    sample_letter_notification,
-    sample_email_notification,
-    sample_notification
+    notify_db_session
 ):
+    service = create_service(service_permissions=[LETTER_TYPE])
+    pdf_service = create_service(service_name='pdf_service', service_permissions=[LETTERS_AS_PDF])
+    pdf_letter_template = create_template(service=pdf_service, template_type=LETTER_TYPE)
+    letter_template = create_template(service=service, template_type=LETTER_TYPE)
+    sms_template = create_template(service=service)
+    email_template = create_template(service=service, template_type=EMAIL_TYPE)
+    letter_notification = create_notification(template=letter_template)
+    pdf_letter_notification = create_notification(template=pdf_letter_template)
+    sms_notification = create_notification(template=sms_template)
+    email_notification = create_notification(template=email_template)
     ret = dao_set_created_live_letter_api_notifications_to_pending()
 
-    assert sample_letter_notification.status == NOTIFICATION_PENDING
-    assert sample_email_notification.status == NOTIFICATION_CREATED
-    assert sample_notification.status == NOTIFICATION_CREATED
-    assert ret == [sample_letter_notification]
+    assert letter_notification.status == NOTIFICATION_PENDING
+    assert email_notification.status == NOTIFICATION_CREATED
+    assert pdf_letter_notification.status == NOTIFICATION_CREATED
+    assert sms_notification.status == NOTIFICATION_CREATED
+    assert ret == [letter_notification]
 
 
-def test_should_ignore_letters_as_pdf(sample_letter_notification):
-    service = create_service(service_permissions=[LETTER_TYPE, 'letters_as_pdf'])
-    template = create_template(service, template_type=LETTER_TYPE)
-    create_notification(template)
+def test_should_ignore_letters_as_pdf(notify_db_session):
+    pdf_service = create_service(service_name='pdf service', service_permissions=[LETTER_TYPE, LETTERS_AS_PDF])
+    pdf_template = create_template(service=pdf_service, template_type=LETTER_TYPE)
+    service = create_service(service_permissions=[LETTER_TYPE])
+    template = create_template(service=service, template_type=LETTER_TYPE)
+
+    notification = create_notification(template)
+    create_notification(pdf_template)
```

Comment:
Should you check that the pdf_template does get set? Or that that tested somewhere else?